packages/@react-spectrum/list/src/ListView.tsx

```diff
@@ -136,6 +151,76 @@ function ListView<T extends object>(props: ListViewProps<T>, ref: DOMRef<HTMLDiv
     // focusable children in the cell.
     focusMode: 'cell'
   }), [state, domRef, direction, collator]);
+
+  let provider = useProvider();
+  let {checkboxProps} = useGridSelectionCheckbox({key: null}, state);
+  let dragState: DraggableCollectionState;
+  if (isListDraggable) {
```

Comment:
so we don't have people accidentally break the rules of hooks, we should have something like this before we get to the if statement

```
  let ref = useRef(isListDraggable);
  let wasDraggable = ref.current;
  let isDraggable = isListDraggable;
  if (wasDraggable !== isDraggable) {
    console.warn(`WARN: something here about must always either supply the hooks or never`);
  }
```

Reply:
looks like you may already be addressing this one